src/events.js

```diff
@@ -13,6 +13,8 @@ export function updateListeners(chart, state, options) {
     if (typeof options[hook] === 'function') {
       state.listened = true;
       state.listeners[hook] = options[hook];
+    } else if (defined(state.listeners[hook])) {
+      delete state.listeners[hook];
```

Comment:
[mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) says:
> If the property which you are trying to delete does not exist, delete will not have any effect and will return true. 

```suggestion
    } else {
      delete state.listeners[hook];
```

Reply:
Yes @kurkle but I have also seen the delete is a costly operation and therefore I want to avoid if not needed.

Reply:
Turns out FireFox is 90% slower deleting non-existent property vs checking its type. So this suggestion was bad.

Reply:
> Turns out FireFox is 90% slower deleting non-existent property vs checking its type. So this suggestion was bad.

Ok, I revert this update